### PR TITLE
go: update gofumpt to 0.1.1

### PR DESCRIPTION
**Pull request by ptrus**
Part of: #3629

### CODE REVIEW

**Review by kostko — APPROVED** [review]

File: docs/setup/prerequisites.md [hunk 1]
```diff
@@ -143,7 +143,7 @@ Core:
   Download and install it with:
 
   ```
-  export GOFUMPT_VERSION=abc0db2c416aca0f60ea33c23c76665f6e7ba0b6
+  export GOFUMPT_VERSION=v0.1.1
   GO111MODULE=on ${OASIS_GO:-go} get mvdan.cc/gofumpt@${GOFUMPT_VERSION}
   GO111MODULE=on ${OASIS_GO:-go} get mvdan.cc/gofumpt/gofumports@${GOFUMPT_VERSION}
```
Comment (tjanez) [inline docs/setup/prerequisites.md:148]: Apparently, `gofumports` is deprecated https://github.com/mvdan/gofumpt/commit/179728cc07226dd410dc11945b2d9917194f9c5a and removed from gofumpt code base https://github.com/mvdan/gofumpt/commit/38fc491470bae6f44e2d38b06277dd95cf1bdf97.

Should we simply remove it here and in the `make fmt` target and rely on people using `gopls` or similar to do proper imports formatting?

Comment (tjanez) [reply docs/setup/prerequisites.md:148]: For golangci-lint, we could add the following [`goimports` configuration](https://golangci-lint.run/usage/linters/#goimports) to check it:

```yaml
linters-settings:
  goimports:
    # Put local imports after 3rd-party packages.
    local-prefixes: github.com/oasisprotocol/oasis-core
```

Comment (tjanez) [inline docs/setup/prerequisites.md:148]: I think we could just replace `gofumports` with `goimports`:

```suggestion
  GO111MODULE=on ${OASIS_GO:-go} get mvdan.cc/gofumpt@v0.1.1
  ${OASIS_GO:-go} get golang.org/x/tools/cmd/goimports
```

And change the `fmt` target in `go/Makefile` to:
```
# Format code.
fmt:
	@$(ECHO) "$(CYAN)*** Running Go formatters...$(OFF)"
	@gofumpt -s -w .
	@goimports -w -local github.com/oasisprotocol/oasis-core .
```

Comment (ptrus) [reply docs/setup/prerequisites.md:148]: Hm ok sure, although `v0.1.1` version of `gofumports` exists and is published. 

Note that there are minor differences in how `gofumports` and `goimports` handle some things:
- https://github.com/golang/go/issues/31646#issue-436515827
- and https://github.com/golang/go/issues/31646#issuecomment-488164935

`gofumports` handled both of these "correctly"

File: docs/setup/prerequisites.md [hunk 2]
```diff
@@ -135,17 +135,16 @@ Core:
   rustup target add x86_64-fortanix-unknown-sgx
   ```
 
-* (**OPTIONAL**) [gofumpt and gofumports].
+* (**OPTIONAL**) [gofumpt and goimports].
```
Comment (tjanez) [inline docs/setup/prerequisites.md:138]:
```suggestion
* (**OPTIONAL**) [gofumpt] and [goimports].
```

File: docs/setup/prerequisites.md [hunk 3]
```diff
@@ -225,7 +224,7 @@ where the code has been checked out.
 [rust-toolchain-precedence]:
   https://github.com/rust-lang/rustup/blob/master/README.md#override-precedence
 [Fortanix Rust EDP]: https://edp.fortanix.com
-[gofumpt and gofumports]: https://github.com/mvdan/gofumpt
+[gofumpt and goimports]: https://github.com/mvdan/gofumpt
```
Comment (tjanez) [inline docs/setup/prerequisites.md:227]:
```suggestion
[gofump]: https://github.com/mvdan/gofumpt
[goimports]: https://pkg.go.dev/golang.org/x/tools/cmd/goimports
```

**Review by tjanez — APPROVED** [review]

**Pull request merged**